eo-maven-plugin/src/main/java/org/eolang/maven/OptimizeMojo.java

```diff
@@ -158,6 +161,25 @@ private XML optimize(final Path file) throws FileNotFoundException {
                 new Home().rel(dir)
             );
         }
+        if (this.failOnError) {
+            trn = trn.with(
+                new StLambda(
+                    xml -> {
+                        final List<String> errors = xml.xpath("//error[@severity='error']/text()");
```

Comment:
@mximp it feels that this code is duplicated. We have this check somewhere else, right? Maybe it's better to make an XSL that does this validation and throws `<xsl:message>`. Then, we can simply add such XSL to the train and that's it.

Reply:
@yegor256 correct: similar check is done in `Sanitized` which called from `TranspileMojo`. It analyses an XMIR from file.
I didn't quite get your idea with another xsl: how would we address `failOnError` flag from within XSL? Passing XSL params via `XSL::with()` method would require almost as much code as we have now to set up new XSL with parameter and passing it.

Reply:
@mximp we just create an XSL stylesheet, which finds `error[@severity=error]` and terminates processing with `<xsl:message>`. Then, we add this XSL to the train with one `.with()` command.